src/urllib3/connectionpool.py

```diff
@@ -1032,7 +1032,7 @@ def _validate_conn(self, conn: HTTPConnection) -> None:
             )
 
 
-def connection_from_url(url: str, **kw: Any) -> ConnectionPool:
+def connection_from_url(url: str, **kw: Any) -> HTTPConnectionPool:
```

Comment:
I don't know if `connection_from_url` wants to keep the option for itself to someday return a `ConnectionPool` which is not an `HTTPConnectionPool`. If it does, then this change would interfere; if not, it's good.

Reply:
++ Yeah these functions aren't what we're recommending anymore, I think it's safe to say we'll only be returning `HTTPConnectionPool`.